### PR TITLE
AM2R: Add IBJ, WJ and Long Beam as unavailable items

### PR DESCRIPTION
**Pull request by Miepee**
Adds those as items, and adjusts the database accordingly.
Preperation for a nearing patcher update that adds these as items.

### CODE REVIEW

File: randovania/games/am2r/logic_database/Golden Temple.txt [hunk 1]
```diff
@@ -421,20 +436,23 @@ Extra - minimap_data: [{'x': 51, 'y': 11}, {'x': 51, 'y': 12}]
...
               # PB and then damage boost
               Damage Boost (Advanced) and Normal Damage ≥ 7 and Can Use Power Bombs
               Any of the following:
-                  Hi-Jump Boots or Mid-Air Morph (Advanced) or Walljump (Beginner) or Can Use Spider Ball
+                  Hi-Jump Boots or Mid-Air Morph (Advanced) or Can Use Spider Ball
                   Charged Bomb Jump (Intermediate) and Can Use Charged Bomb Jump
+                  Walljump (Beginner) and Can Walljump
           # Spring Ball + IBJ
-          Spring Ball and Infinite Bomb Jumping (Intermediate) and Walljump (Intermediate) and Can Use Bombs
+          Spring Ball and Infinite Bomb Jumping (Intermediate) and Walljump (Intermediate) and Can IBJ and Can Use Bombs
```
Comment (JeffGainsNGames) [inline randovania/games/am2r/logic_database/Golden Temple.txt:455]: no `can WJ`

File: randovania/games/am2r/logic_database/Hydro Station.txt [hunk 2]
```diff
@@ -1438,19 +1461,23 @@ Extra - minimap_data: [{'x': 19, 'y': 23}, {'x': 19, 'y': 24}, {'x': 20, 'y': 23
...
-                  Morph Ball and Morph Glide (Intermediate) and Walljump (Advanced)
+                  Morph Ball and Morph Glide (Intermediate) and Walljump (Advanced) and Can Walljump
                   # CBJ method
                   Charged Bomb Jump (Advanced) and Can Use Charged Bomb Jump
                   All of the following:
                       # IBJ method
                       Can Use Bombs
-                      Diagonal Infinite Bomb Jumping (Advanced) or Infinite Bomb Jumping (Advanced)
+                      Any of the following:
+                          Diagonal Infinite Bomb Jumping (Advanced)
```
Comment (JeffGainsNGames) [inline randovania/games/am2r/logic_database/Hydro Station.txt:1479]: should DIBJ have IBJ??

File: randovania/games/am2r/logic_database/Main Caves.txt [hunk 3]
```diff
@@ -558,17 +562,19 @@ Extra - minimap_data: [{'x': 49, 'y': 20}, {'x': 49, 'y': 21}, {'x': 50, 'y': 20
...
-                  Infinite Bomb Jumping (Advanced) and Can Use Bombs
+                  Infinite Bomb Jumping (Advanced) and Can IBJ and Can Use Bombs
                   All of the following:
                       # Glide jump from crumble to crumble
-                      Morph Ball and Morph Glide (Advanced) and Walljump (Beginner)
-                      Power Grip or Walljump (Advanced)
+                      Morph Ball and Morph Glide (Advanced) and Walljump (Beginner) and Can Walljump
+                      Any of the following:
+                          Power Grip
+                          Walljump (Advanced) and Can Walljump
```
Comment (JeffGainsNGames) [inline randovania/games/am2r/logic_database/Main Caves.txt:571]: duplicate `can WJs` not too concerned though

**Review by JeffGainsNGames — COMMENTED** [review]
Please check that zip tricks also check for `can WJ`

**Comment by Miepee** [comment]
zip tricks are accounted for. there are only 3 of them in a5 which i added manually

File: randovania/games/am2r/logic_database/Hydro Station.txt (hunk 2 above)
Comment (Miepee) [reply randovania/games/am2r/logic_database/Hydro Station.txt:1479]: they should, I forgot.

File: randovania/games/am2r/logic_database/Golden Temple.txt (hunk 1 above)
Comment (Miepee) [reply randovania/games/am2r/logic_database/Golden Temple.txt:455]: i have no clue how this happened.

**Comment by Miepee** [comment]
Noticed there were a lot more. After grepping through all tricks for Walljump, IBJ, DBJ and Zip Trick, they *should* now all have the template attached.

**Review by JeffGainsNGames — APPROVED** [review]

**Pull request merged**